infrastructure/firecracker/config.go

```diff
@@ -113,6 +119,40 @@ func WithMicroVM(vm *models.MicroVM) ConfigOption {
 	}
 }
 
+// DefaultKernelCmdLine is the default recommended kernel parameter list.
+//
+// console=ttyS0   [KLN] Output console device and options
+// reboot=k        [KNL] reboot_type=kbd
+// panic=1         [KNL] Kernel behaviour on panic: delay <timeout>
+//                         timeout > 0: seconds before rebooting
+//                         timeout = 0: wait forever
+//                         timeout < 0: reboot immediately
+// pci=off         [X86] don't probe for the PCI bus
+// i8042.noaux     [HW]  Don't check for auxiliary (== mouse) port
+// i8042.nomux     [HW]  Don't check presence of an active multiplexing
+//                       controller
+// i8042.nopnp     [HW]  Don't use ACPIPnP / PnPBIOS to discover KBD/AUX
+//                       controllers
+// i8042.dumbkbd   [HW]  Pretend that controller can only read data from
+//                       keyboard and cannot control its state
+//                       (Don't attempt to blink the leds)
+//
+// Read more:
+// https://www.kernel.org/doc/html/v5.15/admin-guide/kernel-parameters.html
+func DefaultKernelCmdLine() config.KernelCmdLine {
+	return config.KernelCmdLine{
+		"console":       "ttyS0",
+		"reboot":        "k",
+		"panic":         "1",
+		"pci":           "off",
+		"i8042.noaux":   "",
+		"i8042.nomux":   "",
+		"i8042.nopnp":   "",
+		"i8042.dumbkbd": "",
+		"ds":            "nocloud-net;s=http://169.254.169.254/latest/",
```

Comment:
Is it worth adding a comment  about the `ds` arg as well?

Reply:
I tried to find a kernel level documentation about it, but the only one I found is the cloudinit documentation that does not really tell us what is it really only to "just set it and you are good", and that documentation can be found easily searching for `nocloud-net`.

https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html